appimage/action.yml

```diff
@@ -0,0 +1,98 @@
+name: "AppImage"
+description: "Builds a Linux AppImage on Debian 10 (Buster)"
+inputs:
+  executable:
+    description: "Executable file to package into an AppImage."
+    required: true
+  icon-file:
+    description: "Icon file to use for the AppImage."
+    required: true
+  desktop-file:
+    description: "Desktop file to use for the AppImage."
+    required: true
+  appdir:
+    description: "Path to use for the AppImage filesystem."
+    required: true
+
+runs:
+  using: composite
+
+  steps:
+    - name: Build packetry-x86_64.AppDir (Linux)
+      shell: bash
+      run: |
+        DEPLOY_GTK_VERSION=4 linuxdeploy-x86_64.AppImage \
+          --appimage-extract-and-run \
+          --appdir ${{ inputs.appdir }} \
+          --executable=${{ inputs.executable }} \
+          --icon-file ${{ inputs.icon-file }} \
+          --plugin gtk \
+          --desktop-file ${{ inputs.desktop-file }}
+
+    - name: Shrink packetry-x86_64.AppDir (Linux)
+      shell: bash
+      run: |
+        # cd to AppDir lib directory
+        cd $GITHUB_WORKSPACE/${{ inputs.appdir }}/usr/lib
+
+        # remove any system libraries we don't want to include in case
+        # of conflicts with a user's system
+        rm -f libX*
+        rm -f libblkid*
+        rm -f libbsd*
+        rm -f libpixman*
+        rm -f libxcb-render*
+        rm -f libzstd*
+
+        # remove duplicate library files and replace with symlinks
+        fdupes --sameline --order=name . | while read -r lib ; do
+            src=$(echo $lib | rev | cut -d' ' -f1 | rev)
+            dupes=$(echo $lib | rev | cut -d' ' -f2- | rev)
+            rm $dupes
+            for dst in $dupes ; do
+                ln -s $src $dst
+            done
+        done
+
+        # strip all libraries
+        strip ./lib*.*
+
+        # return to workspace directory
+        cd $GITHUB_WORKSPACE
+
+    - name: Gather licenses (Linux)
+      shell: bash
+      run: |
+        # cd to AppDir /usr/share/doc directory
+        cd $GITHUB_WORKSPACE/${{ inputs.appdir }}/usr/share/doc
+
+        # remove any licenses for system libraries we don't distribute
+        rm -rf libblkid1
+        rm -rf libbsd0
+        rm -rf libxau6
+        rm -rf libxcb-render0
+        rm -rf libxcursor1
+        rm -rf libxdamage1
+        rm -rf libxdmcp6
+        rm -rf libxext6
+        rm -rf libxfixes3
+        rm -rf libxi6
+        rm -rf libxinerama1
+        rm -rf libxrandr2
+        rm -rf libxrender1
+        rm -rf libzstd1
+
+        # Return to workspace directory
+        cd $GITHUB_WORKSPACE
+
+    - name: Convert icon to application/x-executable on all themes (Linux)
+      shell: bash
+      run: |
+        cp /dev/null ${{ inputs.appdir }}/usr/share/icons/hicolor/512x512/apps/$(basename ${{ inputs.icon-file }})
```

Comment:
We're including an icon file now (albeit a placeholder one), so why do we still need to do this?

Reply:
Because this is the second step of a multi-step process that starts with `linuxdeploy` needing a valid `.png` as input… 🥲

Basically:

* **Step 1:** `linuxdeploy` + `linuxdeploy-plugin-gtk` populates `packetry.AppDir` 
* **Step 2:** We truncate the `.png` in `packetry.AppDir` to zero bytes.
* **Step 3:** We use `appimagetool` to generate `packetry-x86_64.AppImage` from `packetry.AppDir`.

Reply:
Oh, so is the idea to ensure that it ends up looking like the `application/x-executable` icon in the user's own theme, rather than the specific one we have in `dist/icon.png`?

I'm not sure that really matters, but I guess it's a nice touch if it works.

Reply:
> Oh, so is the idea to ensure that it ends up looking like the `application/x-executable` icon in the user's own theme, rather than the specific one we have in `dist/icon.png`?

💯